extensions/data-plane/data-plane-api/src/main/java/org/eclipse/dataspaceconnector/dataplane/api/transfer/DataPlanePublicApiController.java

```diff
@@ -0,0 +1,113 @@
+ *
+ */
+package org.eclipse.dataspaceconnector.dataplane.api.transfer;
+
+import jakarta.ws.rs.Consumes;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.MultivaluedMap;
+import jakarta.ws.rs.core.Request;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.core.UriInfo;
+import org.eclipse.dataspaceconnector.dataplane.core.util.QueryParamsUtils;
+import org.eclipse.dataspaceconnector.dataplane.spi.manager.DataPlaneManager;
+import org.eclipse.dataspaceconnector.dataplane.spi.pipeline.OutputStreamDataSink;
+import org.eclipse.dataspaceconnector.dataplane.spi.result.TransferResult;
+import org.eclipse.dataspaceconnector.dataplane.spi.security.TokenValidationClient;
+import org.eclipse.dataspaceconnector.spi.iam.ClaimToken;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.response.ResponseStatus;
+import org.eclipse.dataspaceconnector.spi.types.TypeManager;
+import org.eclipse.dataspaceconnector.spi.types.domain.DataAddress;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.DataFlowRequest;
+
+import java.io.ByteArrayOutputStream;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static org.eclipse.dataspaceconnector.dataplane.api.common.ResponseFunctions.internalErrors;
+import static org.eclipse.dataspaceconnector.dataplane.api.common.ResponseFunctions.notAuthorizedErrors;
+import static org.eclipse.dataspaceconnector.dataplane.api.common.ResponseFunctions.validationError;
+import static org.eclipse.dataspaceconnector.dataplane.api.common.ResponseFunctions.validationErrors;
+import static org.eclipse.dataspaceconnector.dataplane.core.schema.DataRequestSchema.METHOD;
+import static org.eclipse.dataspaceconnector.dataplane.core.schema.DataRequestSchema.QUERY_PARAMS;
+
+/**
+ * Controller for synchronous data pull APIs of the data plane.
+ */
+@Consumes({MediaType.APPLICATION_JSON})
+@Produces({MediaType.APPLICATION_JSON})
+@Path("/transfer")
+public class DataPlanePublicApiController {
+
+    private final ExecutorService executorService = Executors.newSingleThreadExecutor();
+    private final TokenValidationClient tokenValidationClient;
+    private final DataPlaneManager dataPlaneManager;
+    private final Monitor monitor;
+    private final TypeManager typeManager;
+
+    public DataPlanePublicApiController(TokenValidationClient tokenValidationClient, DataPlaneManager dataPlaneManager, Monitor monitor, TypeManager typeManager) {
+        this.tokenValidationClient = tokenValidationClient;
+        this.dataPlaneManager = dataPlaneManager;
+        this.monitor = monitor;
+        this.typeManager = typeManager;
+    }
+
+    @GET
+    @Path("validate/{token}")
```

Comment:
wouldn't it be better to pass the token as part of the http headers?
* pathes often get logged and we should avoid to have security information in logfiles.
* also the token could get quite long
* and we could handle various pathes belonging to the same contract in future (query params are not sufficient in every case)
What do you think @bscholtes1A ?